src/ReinforcementLearningCore/src/policies/agent/offline_agent.jl

```diff
@@ -0,0 +1,76 @@
+export OfflineAgent, OfflineBehavior
+
+"""
+    OfflineBehavior(; agent:: Union{<:Agent, Nothing}, steps::Int, reset_condition)
+
+Used to provide an OfflineAgent with a "behavior agent" that will generate the training data
+at the `PreExperimentStage`. If `agent` is `nothing` (by default), does nothing. The `trajectory` of agent should 
+be the same as that of the parent `OfflineAgent`.
+`steps` is the number of data elements to generate, defautls to the capacity of the trajectory.
+`reset_condition` is the episode reset condition for the data generation (defaults to `ResetAtTerminal()`).
+
+The behavior agent will interact with the main environment of the experiment to generate the data.
+"""
+struct OfflineBehavior{A <: Union{<:Agent, Nothing}, R}
```

Comment:
Is `<:Agent` right here or should it be `AbstractAgent`?

Reply:
It must be a classic Agent, otherwise it won't generate data, which is the point.